### PR TITLE
FORGE-2382: Add a serializable parameter to the cdi-new-bean command

### PR DESCRIPTION
**Pull request by agoncal**

### CODE REVIEW

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java [hunk 1]
```diff
@@ -113,6 +118,10 @@ else if (BeanScope.DEPENDENT != scopedValue)
       {
          source.addAnnotation(qualifier.getValue());
       }
+      if (serializable.getValue())
+      {
+         source.addInterface(Serializable.class);
```
Comment (gastaldi) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java]: serialVersionUID should be added too I suppose?

Comment (agoncal) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java]: Hum.... and you will set it to 1L ? Not a good practice.

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/BeanScope.java [hunk 2]
```diff
@@ -11,22 +11,28 @@
  */
 public enum BeanScope
 {
-   DEPENDENT(""),
-   APPLICATION("javax.enterprise.context.ApplicationScoped"),
-   SESSION("javax.enterprise.context.SessionScoped"),
-   CONVERSATION("javax.enterprise.context.ConversationScoped"),
-   REQUEST("javax.enterprise.context.RequestScoped"),
-   CUSTOM(null);
+   DEPENDENT("", false),
```
Comment (agoncal) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/BeanScope.java]: I realized that when you have a session, application or conversation scope bean, it must be serializable.

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java [hunk 3]
```diff
@@ -100,6 +101,9 @@ public JavaClassSource decorateSource(UIExecutionContext context, Project projec
       else if (BeanScope.DEPENDENT != scopedValue)
       {
          source.addAnnotation(scopedValue.getAnnotation());
+         if (scopedValue.isSerializable()) {
+            source.addInterface(Serializable.class);
```
Comment (gastaldi) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java]: I think it´s fine to add a serialVersionUID = 1 field to this class. The IDE warnings are a bit annoying to have.
We could introduce a method to calculate the serialVersion based on http://docs.oracle.com/javase/8/docs/platform/serialization/spec/class.html#a4100 in Roaster later

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java [hunk 4]
```diff
@@ -100,6 +101,12 @@ public JavaClassSource decorateSource(UIExecutionContext context, Project projec
       else if (BeanScope.DEPENDENT != scopedValue)
       {
          source.addAnnotation(scopedValue.getAnnotation());
+         if (scopedValue.isSerializable())
+         {
+            source.addInterface(Serializable.class);
+            source.addField().setPrivate().setStatic(true).setFinal(true).setName("serialVersionUID").setType("long")
+                     .setLiteralInitializer("1L");
+         }
```
Comment (agoncal) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/cdi/ui/CDINewBeanCommand.java]: Here is the serialVersionUID ;o)

**Pull request merged**